test/ng/parseSpec.js

```diff
@@ -2973,6 +2973,172 @@ describe('parser', function() {
           expect(watcherCalls).toBe(1);
         }));
 
+        it('should not reevaluate filters in literals with non-primitive input that does support valueOf()',
```

Comment:
can we use some describe blocks to simplify the test descriptions here:

```
describe('containing literals', function() {
  describe('with non-primitive inputs', function() {
    describe('that support valueOf()', function() {
      it('should not reevaluate filters, inputs do not change', ...);
      it('should reevaluate filters, when an input valueOf() result changes', ...);
      it('should not reevaluate filters, when the input changes but valueOf() does not ', ...);
    });
  });
  describe('with non-primitive inputs', function() {
  });
});
```

you get the idea...